src/trans.rs

```diff
@@ -331,41 +353,165 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
                 self.fun_types.insert(self.sig.clone(), ty);
             }
 
+            // Create the function prologue
+            // TODO: the epilogue and prologue are not always necessary
+            let read_sp = BinaryenLoad(self.module, 4, 0, 0, 0, BinaryenInt32(),
+                                       BinaryenConst(self.module, BinaryenLiteralInt32(0)));
```

Comment:
Awesome! Yeah, I was thinking factoring the stack manipulation code into a common location would be a good idea, but it sounds like you're already on it.